### PR TITLE
[gosrc2cpg] - Generic type return handling

### PR DESCRIPTION
**Pull request by ankit-privado**
Added support to handle the ast creation for return generic type method. Also added test for multiple generic type parameter.

### CODE REVIEW

File: joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstForFunctionsCreator.scala [hunk 1]
```diff
@@ -45,14 +45,14 @@ trait AstForFunctionsCreator(implicit withSchemaValidation: ValidationMode) { th
...
+      getReturnType(funcDecl.json(ParserKeys.Type), genericTypeMethodMap).headOption
         .getOrElse(("", methodReturnNode(funcDecl, Defines.voidTypeName)))
-    val templateParams       = ""
-    val params               = funcDecl.json(ParserKeys.Type)(ParserKeys.Params)(ParserKeys.List)
-    val genericTypeMethodMap = processTypeParams(funcDecl.json(ParserKeys.Type))
+    val templateParams = ""
+    val params         = funcDecl.json(ParserKeys.Type)(ParserKeys.Params)(ParserKeys.List)
     val signature =
-      s"$fullname$templateParams(${parameterSignature(params, genericTypeMethodMap)})$returnTypeStr"
+      s"$fullname$templateParams(${parameterSignature(params, genericTypeMethodMap)})($returnTypeStr)"
```
Comment (pandurangpatil) [inline joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstForFunctionsCreator.scala:55]: `(`$returnTypeStr`)` this addition of bracket will break many unit tests inside MethodTest

Comment (ankit-privado) [reply joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstForFunctionsCreator.scala:55]: @pandurangpatil should i change all those unit test?

Comment (pandurangpatil) [reply joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstForFunctionsCreator.scala:55]: change the unit tests

Comment (pandurangpatil) [reply joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstForFunctionsCreator.scala:55]: @ankit-privado remove the brackets and update only your unit tests

Comment (ankit-privado) [reply joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstForFunctionsCreator.scala:55]: okay sure @pandurangpatil

**Review by DavidBakerEffendi — APPROVED** [review]

**Pull request merged**